src/compile/mark/valueref.ts

```diff
@@ -102,7 +102,7 @@ export function midPoint(channel: Channel, channelDef: ChannelDef<string>, scale
       if (channelDef.bin) {
         // Use middle only for x an y to place marks in the center between start and end of the bin range.
         // We do not use the mid point for other channels (e.g. size) so that properties of legends and marks match.
-        if (contains(['x', 'y'], channel)) {
+        if (contains(['x', 'y'], channel) && channelDef.type === 'quantitative') {
```

Comment:
Good catch.   Maybe add comment that we only impute bin Q but not bin T?

Reply:
I think the code says that already.

Reply:
sort of, but it doesn't say why and it's kinda complicated so doesn't hurt to note.

Reply:
I don't understand why this predicate below for using range doesn't match `binRequiresRange`.  
```
return fieldRef(channelDef, scaleName, isBinScale(scale.get('type')) ? {} : {binSuffix: 'range'});
```

Should it be 
```
return fieldRef(channelDef, scaleName, binRequiresRange(channelDef, channel) ? {binSuffix: 'range'} : {});
``` 
instead?  (I'm not 100% sure)

Reply:
The code doesn't produce different results in any cases I can construct and that make sense.  Changed.